File: cpp/design-patterns/flyweight/factory/tree_factory.cpp
class TreeFactory {
    static std::unordered_map<std::string, TreeType*> types;
public:
    static TreeType* getTreeType(const std::string& name, const std::string& color) {
        std::string key = name + "_" + color;
        if (types.find(key) == types.end()) {
            types[key] = new TreeType(name, color);
        }
        return types[key];
    }
};

File: java/design-patterns/flyweight/factory/TreeFactory.java
public class TreeFactory {
    private static final Map<String, TreeType> types = new HashMap<>();

    public static TreeType getTreeType(String name, String color) {
        String key = name + "_" + color;
        if (!types.containsKey(key)) {
            types.put(key, new TreeType(name, color));
        }
        return types.get(key);
    }
}